etna/datasets/tsdataset.py

```diff
@@ -198,6 +199,24 @@ def _cast_segment_to_str(df: pd.DataFrame) -> pd.DataFrame:
         df.columns = pd.MultiIndex.from_frame(columns_frame)
         return df
 
+    @staticmethod
+    def _check_timestamp_type(df: pd.DataFrame):
+        if df.index.dtype == "int":
+            warnings.warn(
+                "Timestamp contains numeric values, which can lead to unexpected results if freq is not None."
+            )
+            return True
+        return False
+
+    @staticmethod
+    def _cast_index_to_datetime(df: pd.DataFrame, freq: str) -> pd.DataFrame:
```

Comment:
I don't understand the meaning of this casting. I expected it to:
- Call raise warning if index is numeric
- Call `df.index = pd.to_datetime(df.index)`

We should call this method in `_prepare_df` and in `__init__` on line 153 (in creating `df_exog`).